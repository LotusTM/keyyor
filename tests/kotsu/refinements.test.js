/* eslint-env jest */

import t from 'tcomb'
import r from '../../modules/refinements'

describe('Tcomb refinement', () => {
  describe('False', () => {
    it('should pass `false`', () => {
      expect(r.False(false)).toMatchSnapshot()
    })
    it('should error on everything except `false`', () => {
      expect(() => r.False(true)).toThrowErrorMatchingSnapshot()
      expect(() => r.False('string')).toThrowErrorMatchingSnapshot()
      expect(() => r.False(-1)).toThrowErrorMatchingSnapshot()
      expect(() => r.False(1)).toThrowErrorMatchingSnapshot()
      expect(() => r.False(-1.555)).toThrowErrorMatchingSnapshot()
      expect(() => r.False(1.555)).toThrowErrorMatchingSnapshot()
      expect(() => r.False(0)).toThrowErrorMatchingSnapshot()
      expect(() => r.False(null)).toThrowErrorMatchingSnapshot()
      expect(() => r.False(undefined)).toThrowErrorMatchingSnapshot()
    })
  })

  describe('Absoluteurl', () => {
    it('should pass absolute url', () => {
      expect(r.Absoluteurl('http://test.com')).toMatchSnapshot()
      expect(r.Absoluteurl('https://test.com')).toMatchSnapshot()
      expect(r.Absoluteurl('//test.com')).toMatchSnapshot()
    })
    it('should error on everything except absolute url', () => {
      expect(() => r.Absoluteurl(true)).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl('/url')).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl('url/')).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl('url')).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl(-1)).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl(1)).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl(-1.555)).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl(1.555)).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl(0)).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl(null)).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl(undefined)).toThrowErrorMatchingSnapshot()
    })
  })

  describe('Imagepath', () => {
    it('should pass path with image file', () => {
      expect(r.Imagepath('file.jpg')).toMatchSnapshot()
      expect(r.Imagepath('file.jpeg')).toMatchSnapshot()
      expect(r.Imagepath('file.png')).toMatchSnapshot()
      expect(r.Imagepath('file.gif')).toMatchSnapshot()
      expect(r.Imagepath('file.svg')).toMatchSnapshot()
    })
    it('should error on everything except path with image file', () => {
      expect(() => r.Imagepath(true)).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath('http://test.com/imagelikepath.svg/')).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath('http://test.com')).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath('/url')).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath('url/')).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath('url')).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath(-1)).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath(1)).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath(-1.555)).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath(1.555)).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath(0)).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath(null)).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath(undefined)).toThrowErrorMatchingSnapshot()
    })
  })

  describe('Handle', () => {
    it('should pass handle', () => {
      expect(r.Handle('@test')).toMatchSnapshot()
    })
    it('should error on everything except handle', () => {
      expect(() => r.Handle(true)).toThrowErrorMatchingSnapshot()
      expect(() => r.Handle('http://@test/')).toThrowErrorMatchingSnapshot()
      expect(() => r.Handle('http://test.com')).toThrowErrorMatchingSnapshot()
      expect(() => r.Handle('@url/test')).toThrowErrorMatchingSnapshot()
      expect(() => r.Handle('@url:test')).toThrowErrorMatchingSnapshot()
      expect(() => r.Handle('@url\\test')).toThrowErrorMatchingSnapshot()
      expect(() => r.Handle('/url')).toThrowErrorMatchingSnapshot()
      expect(() => r.Handle('url/')).toThrowErrorMatchingSnapshot()
      expect(() => r.Handle('url')).toThrowErrorMatchingSnapshot()
      expect(() => r.Handle(-1)).toThrowErrorMatchingSnapshot()
      expect(() => r.Handle(1)).toThrowErrorMatchingSnapshot()
      expect(() => r.Handle(-1.555)).toThrowErrorMatchingSnapshot()
      expect(() => r.Handle(1.555)).toThrowErrorMatchingSnapshot()
      expect(() => r.Handle(0)).toThrowErrorMatchingSnapshot()
      expect(() => r.Handle(null)).toThrowErrorMatchingSnapshot()
      expect(() => r.Handle(undefined)).toThrowErrorMatchingSnapshot()
    })
  })

  describe('Maxlength', () => {
    it('should pass string with less or equal length', () => {
      expect(r.Maxlength(3)(t.String)('12')).toMatchSnapshot()
      expect(r.Maxlength(3)(t.String)('123')).toMatchSnapshot()
    })
    it('should pass array with less or equal length', () => {
      expect(r.Maxlength(3)(t.list(t.Any))([1, 2])).toMatchSnapshot()
      expect(r.Maxlength(3)(t.list(t.Any))([1, 2, 3])).toMatchSnapshot()
    })
    it('should error on not tcomb type', () => {
      expect(() => r.Maxlength(3)('nope')('12345678')).toThrowErrorMatchingSnapshot()
    })
    it('should error on not `number` max', () => {
      expect(() => r.Maxlength('nope')(t.String)('12345678')).toThrowErrorMatchingSnapshot()
    })
    it('should error on long strings', () => {
      expect(() => r.Maxlength(3)(t.String)('1234')).toThrowErrorMatchingSnapshot()
    })
    it('should error on long arrays', () => {
      expect(() => r.Maxlength(3)(t.list(t.Any))([1, 2, 3, 4])).toThrowErrorMatchingSnapshot()
    })
  })

  describe('Date', () => {
    it('should pass ISO 8601 date', () => {
      expect(r.Date('2010-01-01T05:06:07')).toMatchSnapshot()
    })
    it('should error on non-ISO 8601 date', () => {
      expect(() => r.Date(1)).toThrowErrorMatchingSnapshot()
      expect(() => r.Date('test')).toThrowErrorMatchingSnapshot()
      expect(() => r.Date('2012')).toThrowErrorMatchingSnapshot()
      expect(() => r.Date('2012-30-30')).toThrowErrorMatchingSnapshot()
      expect(() => r.Date('2010-01-01T05:')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('MsDate', () => {
    it('should pass date in ms since January 1, 1970', () => {
      expect(r.MsDate(-200)).toMatchSnapshot()
      expect(r.MsDate(0)).toMatchSnapshot()
      expect(r.MsDate(200)).toMatchSnapshot()
    })
    it('should error on non-ISO 8601 date', () => {
      expect(() => r.MsDate('2010-01-01T05:06:07')).toThrowErrorMatchingSnapshot()
      expect(() => r.MsDate(22222222222222222222222)).toThrowErrorMatchingSnapshot()
      expect(() => r.MsDate('test')).toThrowErrorMatchingSnapshot()
      expect(() => r.MsDate('2012')).toThrowErrorMatchingSnapshot()
      expect(() => r.MsDate('2012-30-30')).toThrowErrorMatchingSnapshot()
      expect(() => r.MsDate('2010-01-01T05:')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('DigitString', () => {
    it('should pass strings with digits only', () => {
      expect(r.DigitString('23123')).toMatchSnapshot()
    })
    it('should error strings with non-digits', () => {
      expect(() => r.DigitString(1)).toThrowErrorMatchingSnapshot()
      expect(() => r.DigitString('test')).toThrowErrorMatchingSnapshot()
      expect(() => r.DigitString('2012-30-30')).toThrowErrorMatchingSnapshot()
      expect(() => r.DigitString('2010-01-01T05:')).toThrowErrorMatchingSnapshot()
    })
  })

  describe('EqualKeyAndProp', () => {
    const EqualKeysAndId = r.EqualKeyAndProp('id')(t.dict(t.String, t.Any, 'Testdata'))

    it('should pass object with equal keys and property value', () => {
      expect(EqualKeysAndId({ 235: { id: '235' } })).toMatchSnapshot()
      expect(EqualKeysAndId({ test: { id: 'test' } })).toMatchSnapshot()
    })
    it('should error on everything except object with equal keys and property value', () => {
      expect(() => EqualKeysAndId({ 235: { id: 235 } })).toThrowErrorMatchingSnapshot()
      expect(() => EqualKeysAndId({ 235: { id: 111 } })).toThrowErrorMatchingSnapshot()
      expect(() => EqualKeysAndId({ test: { id: 'nope' } })).toThrowErrorMatchingSnapshot()
      // @todo This one throws only first error, should not stop
      // expect(() => EqualKeysAndId({ id1: { id: 'nope' }, id2: { id: 'nopetwice' } })).toThrowErrorMatchingSnapshot()
      expect(() => EqualKeysAndId({ test: { other: 'property' } })).toThrowErrorMatchingSnapshot()
      expect(() => EqualKeysAndId(true)).toThrowErrorMatchingSnapshot()
      expect(() => EqualKeysAndId('string')).toThrowErrorMatchingSnapshot()
      expect(() => EqualKeysAndId(111)).toThrowErrorMatchingSnapshot()
      expect(() => EqualKeysAndId(0)).toThrowErrorMatchingSnapshot()
      expect(() => EqualKeysAndId(null)).toThrowErrorMatchingSnapshot()
      expect(() => EqualKeysAndId(undefined)).toThrowErrorMatchingSnapshot()
      expect(() => r.EqualKeyAndProp('id')('string')).toThrowErrorMatchingSnapshot()
      expect(() => r.EqualKeyAndProp('id')(111)).toThrowErrorMatchingSnapshot()
    })
  })
})
