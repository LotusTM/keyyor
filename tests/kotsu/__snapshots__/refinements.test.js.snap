// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 1`] = `"[tcomb] Invalid value true supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 2`] = `"[tcomb] Invalid value \\"/url\\" supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 3`] = `"[tcomb] Invalid value \\"url/\\" supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 4`] = `"[tcomb] Invalid value \\"url\\" supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 5`] = `"[tcomb] Invalid value -1 supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 6`] = `"[tcomb] Invalid value 1 supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 7`] = `"[tcomb] Invalid value -1.555 supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 8`] = `"[tcomb] Invalid value 1.555 supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 9`] = `"[tcomb] Invalid value 0 supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 10`] = `"[tcomb] Invalid value null supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 11`] = `"[tcomb] Invalid value undefined supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should pass absolute url 1`] = `"http://test.com"`;

exports[`Tcomb refinement Absoluteurl should pass absolute url 2`] = `"https://test.com"`;

exports[`Tcomb refinement Absoluteurl should pass absolute url 3`] = `"//test.com"`;

exports[`Tcomb refinement Date should error on non-ISO 8601 date 1`] = `"[tcomb] Invalid value 1 supplied to ISO 8601 Date"`;

exports[`Tcomb refinement Date should error on non-ISO 8601 date 2`] = `"[tcomb] Invalid value \\"test\\" supplied to ISO 8601 Date"`;

exports[`Tcomb refinement Date should error on non-ISO 8601 date 3`] = `"[tcomb] Invalid value \\"2012\\" supplied to ISO 8601 Date"`;

exports[`Tcomb refinement Date should error on non-ISO 8601 date 4`] = `"[tcomb] Invalid value \\"2012-30-30\\" supplied to ISO 8601 Date"`;

exports[`Tcomb refinement Date should error on non-ISO 8601 date 5`] = `"[tcomb] Invalid value \\"2010-01-01T05:\\" supplied to ISO 8601 Date"`;

exports[`Tcomb refinement Date should pass ISO 8601 date 1`] = `"2010-01-01T05:06:07"`;

exports[`Tcomb refinement DigitString should error strings with non-digits 1`] = `"[tcomb] Invalid value 1 supplied to String with digits only"`;

exports[`Tcomb refinement DigitString should error strings with non-digits 2`] = `"[tcomb] Invalid value \\"test\\" supplied to String with digits only"`;

exports[`Tcomb refinement DigitString should error strings with non-digits 3`] = `"[tcomb] Invalid value \\"2012-30-30\\" supplied to String with digits only"`;

exports[`Tcomb refinement DigitString should error strings with non-digits 4`] = `"[tcomb] Invalid value \\"2010-01-01T05:\\" supplied to String with digits only"`;

exports[`Tcomb refinement DigitString should pass strings with digits only 1`] = `"23123"`;

exports[`Tcomb refinement EqualKeyAndProp should error on everything except object with equal keys and property value 1`] = `"[tcomb] ivalid value \`235\`<number> supplied to \`Testdata/235/id\` property: expected equal to key value \`235\`<string>"`;

exports[`Tcomb refinement EqualKeyAndProp should error on everything except object with equal keys and property value 2`] = `"[tcomb] ivalid value \`111\`<number> supplied to \`Testdata/235/id\` property: expected equal to key value \`235\`<string>"`;

exports[`Tcomb refinement EqualKeyAndProp should error on everything except object with equal keys and property value 3`] = `"[tcomb] ivalid value \`nope\`<string> supplied to \`Testdata/test/id\` property: expected equal to key value \`test\`<string>"`;

exports[`Tcomb refinement EqualKeyAndProp should error on everything except object with equal keys and property value 4`] = `"[tcomb] ivalid value \`undefined\`<undefined> supplied to \`Testdata/test/id\` property: expected equal to key value \`test\`<string>"`;

exports[`Tcomb refinement EqualKeyAndProp should error on everything except object with equal keys and property value 5`] = `"[tcomb] Invalid value true supplied to Testdata"`;

exports[`Tcomb refinement EqualKeyAndProp should error on everything except object with equal keys and property value 6`] = `"[tcomb] Invalid value \\"string\\" supplied to Testdata"`;

exports[`Tcomb refinement EqualKeyAndProp should error on everything except object with equal keys and property value 7`] = `"[tcomb] Invalid value 111 supplied to Testdata"`;

exports[`Tcomb refinement EqualKeyAndProp should error on everything except object with equal keys and property value 8`] = `"[tcomb] Invalid value 0 supplied to Testdata"`;

exports[`Tcomb refinement EqualKeyAndProp should error on everything except object with equal keys and property value 9`] = `"[tcomb] Invalid value null supplied to Testdata"`;

exports[`Tcomb refinement EqualKeyAndProp should error on everything except object with equal keys and property value 10`] = `"[tcomb] Invalid value undefined supplied to Testdata"`;

exports[`Tcomb refinement EqualKeyAndProp should error on everything except object with equal keys and property value 11`] = `"[tcomb] Invalid value \\"string\\" supplied to Type"`;

exports[`Tcomb refinement EqualKeyAndProp should error on everything except object with equal keys and property value 12`] = `"[tcomb] Invalid value 111 supplied to Type"`;

exports[`Tcomb refinement EqualKeyAndProp should pass object with equal keys and property value 1`] = `
Object {
  "235": Object {
    "id": "235",
  },
}
`;

exports[`Tcomb refinement EqualKeyAndProp should pass object with equal keys and property value 2`] = `
Object {
  "test": Object {
    "id": "test",
  },
}
`;

exports[`Tcomb refinement False should error on everything except \`false\` 1`] = `"[tcomb] Invalid value true supplied to False"`;

exports[`Tcomb refinement False should error on everything except \`false\` 2`] = `"[tcomb] Invalid value \\"string\\" supplied to False"`;

exports[`Tcomb refinement False should error on everything except \`false\` 3`] = `"[tcomb] Invalid value -1 supplied to False"`;

exports[`Tcomb refinement False should error on everything except \`false\` 4`] = `"[tcomb] Invalid value 1 supplied to False"`;

exports[`Tcomb refinement False should error on everything except \`false\` 5`] = `"[tcomb] Invalid value -1.555 supplied to False"`;

exports[`Tcomb refinement False should error on everything except \`false\` 6`] = `"[tcomb] Invalid value 1.555 supplied to False"`;

exports[`Tcomb refinement False should error on everything except \`false\` 7`] = `"[tcomb] Invalid value 0 supplied to False"`;

exports[`Tcomb refinement False should error on everything except \`false\` 8`] = `"[tcomb] Invalid value null supplied to False"`;

exports[`Tcomb refinement False should error on everything except \`false\` 9`] = `"[tcomb] Invalid value undefined supplied to False"`;

exports[`Tcomb refinement False should pass \`false\` 1`] = `false`;

exports[`Tcomb refinement Handle should error on everything except handle 1`] = `"[tcomb] Invalid value true supplied to Handle"`;

exports[`Tcomb refinement Handle should error on everything except handle 2`] = `"[tcomb] Invalid value \\"http://@test/\\" supplied to Handle"`;

exports[`Tcomb refinement Handle should error on everything except handle 3`] = `"[tcomb] Invalid value \\"http://test.com\\" supplied to Handle"`;

exports[`Tcomb refinement Handle should error on everything except handle 4`] = `"[tcomb] Invalid value \\"@url/test\\" supplied to Handle"`;

exports[`Tcomb refinement Handle should error on everything except handle 5`] = `"[tcomb] Invalid value \\"@url:test\\" supplied to Handle"`;

exports[`Tcomb refinement Handle should error on everything except handle 6`] = `"[tcomb] Invalid value \\"@url\\\\\\\\test\\" supplied to Handle"`;

exports[`Tcomb refinement Handle should error on everything except handle 7`] = `"[tcomb] Invalid value \\"/url\\" supplied to Handle"`;

exports[`Tcomb refinement Handle should error on everything except handle 8`] = `"[tcomb] Invalid value \\"url/\\" supplied to Handle"`;

exports[`Tcomb refinement Handle should error on everything except handle 9`] = `"[tcomb] Invalid value \\"url\\" supplied to Handle"`;

exports[`Tcomb refinement Handle should error on everything except handle 10`] = `"[tcomb] Invalid value -1 supplied to Handle"`;

exports[`Tcomb refinement Handle should error on everything except handle 11`] = `"[tcomb] Invalid value 1 supplied to Handle"`;

exports[`Tcomb refinement Handle should error on everything except handle 12`] = `"[tcomb] Invalid value -1.555 supplied to Handle"`;

exports[`Tcomb refinement Handle should error on everything except handle 13`] = `"[tcomb] Invalid value 1.555 supplied to Handle"`;

exports[`Tcomb refinement Handle should error on everything except handle 14`] = `"[tcomb] Invalid value 0 supplied to Handle"`;

exports[`Tcomb refinement Handle should error on everything except handle 15`] = `"[tcomb] Invalid value null supplied to Handle"`;

exports[`Tcomb refinement Handle should error on everything except handle 16`] = `"[tcomb] Invalid value undefined supplied to Handle"`;

exports[`Tcomb refinement Handle should pass handle 1`] = `"@test"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 1`] = `"[tcomb] Invalid value true supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 2`] = `"[tcomb] Invalid value \\"http://test.com/imagelikepath.svg/\\" supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 3`] = `"[tcomb] Invalid value \\"http://test.com\\" supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 4`] = `"[tcomb] Invalid value \\"/url\\" supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 5`] = `"[tcomb] Invalid value \\"url/\\" supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 6`] = `"[tcomb] Invalid value \\"url\\" supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 7`] = `"[tcomb] Invalid value -1 supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 8`] = `"[tcomb] Invalid value 1 supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 9`] = `"[tcomb] Invalid value -1.555 supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 10`] = `"[tcomb] Invalid value 1.555 supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 11`] = `"[tcomb] Invalid value 0 supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 12`] = `"[tcomb] Invalid value null supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 13`] = `"[tcomb] Invalid value undefined supplied to Image file"`;

exports[`Tcomb refinement Imagepath should pass path with image file 1`] = `"file.jpg"`;

exports[`Tcomb refinement Imagepath should pass path with image file 2`] = `"file.jpeg"`;

exports[`Tcomb refinement Imagepath should pass path with image file 3`] = `"file.png"`;

exports[`Tcomb refinement Imagepath should pass path with image file 4`] = `"file.gif"`;

exports[`Tcomb refinement Imagepath should pass path with image file 5`] = `"file.svg"`;

exports[`Tcomb refinement Maxlength should error on long arrays 1`] = `
"[tcomb] Invalid value [
  1,
  2,
  3,
  4
] supplied to Array<Any>, Maxlength 3"
`;

exports[`Tcomb refinement Maxlength should error on long strings 1`] = `"[tcomb] Invalid value \\"1234\\" supplied to String, Maxlength 3"`;

exports[`Tcomb refinement Maxlength should error on not \`number\` max 1`] = `"[tcomb] Invalid value \\"nope\\" supplied to Number"`;

exports[`Tcomb refinement Maxlength should error on not tcomb type 1`] = `"[tcomb] Invalid value \\"nope\\" supplied to Type"`;

exports[`Tcomb refinement Maxlength should pass array with less or equal length 1`] = `
Array [
  1,
  2,
]
`;

exports[`Tcomb refinement Maxlength should pass array with less or equal length 2`] = `
Array [
  1,
  2,
  3,
]
`;

exports[`Tcomb refinement Maxlength should pass string with less or equal length 1`] = `"12"`;

exports[`Tcomb refinement Maxlength should pass string with less or equal length 2`] = `"123"`;

exports[`Tcomb refinement MsDate should error on non-ISO 8601 date 1`] = `"[tcomb] Invalid value \\"2010-01-01T05:06:07\\" supplied to Date in ms since January 1, 1970"`;

exports[`Tcomb refinement MsDate should error on non-ISO 8601 date 2`] = `"[tcomb] Invalid value 2.2222222222222223e+22 supplied to Date in ms since January 1, 1970"`;

exports[`Tcomb refinement MsDate should error on non-ISO 8601 date 3`] = `"[tcomb] Invalid value \\"test\\" supplied to Date in ms since January 1, 1970"`;

exports[`Tcomb refinement MsDate should error on non-ISO 8601 date 4`] = `"[tcomb] Invalid value \\"2012\\" supplied to Date in ms since January 1, 1970"`;

exports[`Tcomb refinement MsDate should error on non-ISO 8601 date 5`] = `"[tcomb] Invalid value \\"2012-30-30\\" supplied to Date in ms since January 1, 1970"`;

exports[`Tcomb refinement MsDate should error on non-ISO 8601 date 6`] = `"[tcomb] Invalid value \\"2010-01-01T05:\\" supplied to Date in ms since January 1, 1970"`;

exports[`Tcomb refinement MsDate should pass date in ms since January 1, 1970 1`] = `-200`;

exports[`Tcomb refinement MsDate should pass date in ms since January 1, 1970 2`] = `0`;

exports[`Tcomb refinement MsDate should pass date in ms since January 1, 1970 3`] = `200`;
